packages/keystone/lib/Keystone/index.js

```diff
@@ -74,7 +74,9 @@ module.exports = class Keystone {
   createList(key, config, { isAuxList = false } = {}) {
     const { getListByKey, adapters } = this;
     const adapterName = config.adapterName || this.defaultAdapter;
-    const list = new List(key, config, {
+    const compose = fns => o => fns.reduce((acc, fn) => fn(acc), o);
+
+    const list = new List(key, compose(config.plugins || []), config, {
```

Comment:
We should also have option to run this for Keystone level plugins

Reply:
@gautamsi that's an interesting idea, and it might come in the future. This PR is purely for setting up a "blessed" API to enhance field config.

Anything more than what's here already is out of scope for now.

Reply:
Should this have executed the composed function?

```suggestion
     const list = new List(key, compose(config.plugins || [])(config), {
```

Reply:
Hah, yep, 💯